File: scala/src/main/scala/fr/xebia/codeelevator/ElevatorEngine.scala
package fr.xebia.codeelevator

import Direction.Direction
import Command.{Command, NOTHING}

class ElevatorEngine {
  def nextCommand: Command = {
    println(s"nextCommand: $NOTHING")
    NOTHING
  }

  def call(atFloor: Int, to: Direction): Unit = {
    println(s"call($atFloor, $to)")
  }

  def go(floorToGo: Int): Unit = {
    println(s"go($floorToGo)")
  }

  def userHasEntered(): Unit = {
    println(s"userHasEntered")
  }

  def userHasExited(): Unit = {
    println(s"userHasExited")
  }

  def reset(cause: String): Unit = {
    println(s"reset($cause)")
  }
}

File: csharp/ElevatorEngine.cs
using System;
using System.ComponentModel;

namespace csharp
{
    public class ElevatorEngine
    {
        public Command NextCommand()
        {
            Console.WriteLine("NextCommand : {0}", Command.Nothing);
            return Command.Nothing;
        }

        public void Call(int atFloor, Direction to)
        {
            Console.WriteLine("Call : ({0}, {1})", atFloor, to);
        }

        public void Go(int floorToGo)
        {
            Console.WriteLine("Go : ({0})", floorToGo);
        }

        public void UserHasEntered()
        {
            Console.WriteLine("UserHasEntered");
        }

        public void UserHasExited()
        {
            Console.WriteLine("UserHasExited");
        }

        public void Reset(string cause)
        {
            Console.WriteLine("reset({0})", cause);
        }
    }
}